frontend/packages/dev-console/src/components/topology/nodeContextMenu.tsx

```diff
@@ -23,34 +23,56 @@ const onKebabOptionClick = (option: KebabOption) => {
   }
 };
 
-const createMenuItems = (actions: KebabMenuOption[]) =>
-  actions.map((option) =>
+const createMenuItems = (
+  actions: KebabMenuOption[],
+  onOptionClick?: (item: KebabOption) => void,
+) => {
+  return actions.map((option) =>
     isKebabSubMenu(option) ? (
       <ContextSubMenuItem label={option.label} key={option.label}>
         {createMenuItems(option.children)}
       </ContextSubMenuItem>
     ) : (
       <ContextMenuItem
         key={option.label}
-        component={<KebabItem option={option} onClick={() => onKebabOptionClick(option)} />}
+        component={
+          <KebabItem
+            option={option}
+            onClick={() => (onOptionClick ? onOptionClick(option) : onKebabOptionClick(option))}
+          />
+        }
       />
     ),
   );
+};
 
 export const workloadContextMenu = (element: Node) =>
   createMenuItems(kebabOptionsToMenu(workloadActions(element.getData())));
 
-export const groupContextMenu = (element: Node) => {
+export const groupContextMenu = (
+  element: Node,
+  connectorDropMenuCallback?: (item: KebabOption) => void,
+) => {
   const applicationData: TopologyApplicationObject = {
     id: element.getId(),
     name: element.getLabel(),
     resources: element.getChildren().map((node: GraphElement) => node.getData()),
   };
 
-  return createMenuItems(kebabOptionsToMenu(groupActions(applicationData)));
+  return connectorDropMenuCallback
+    ? createMenuItems(groupActions(applicationData, true), connectorDropMenuCallback)
+    : createMenuItems(kebabOptionsToMenu(groupActions(applicationData)));
 };
 export const nodeContextMenu = (element: Node) =>
   createMenuItems(kebabOptionsToMenu(nodeActions(element.getData())));
 
-export const graphContextMenu = (element: Graph) =>
-  createMenuItems(kebabOptionsToMenu(graphActions(element.getController().getElements())));
+export const graphContextMenu = (
+  element: Graph,
+  connectorDropMenuCallback?: (item: KebabOption) => void,
```

Comment:
Lots of changes will need to be made to this file to support my suggestion about improving the href generation in the actions utils.

```suggestion
  contextSource?: <type>,
```